examples/k8s_spark_plugin/k8s_spark_plugin/dataframe_passing.py

```diff
@@ -3,29 +3,53 @@
 #
 # # Converting a Spark DataFrame to a Pandas DataFrame
 #
-# This example shows how a Spark dataset can be returned from a Flyte task and consumed as a pandas DataFrame.
+# This example shows the process of returning a Spark dataset from a Flyte task
+# and then utilizing it as a Pandas DataFrame.
 
 # %% [markdown]
-# First, we import the libraries.
+# To begin, import the libraries.
 # %%
 import flytekit
 import pandas
-from flytekit import Resources, kwtypes, task, workflow
+from flytekit import ImageSpec, Resources, kwtypes, task, workflow
 from flytekit.types.structured.structured_dataset import StructuredDataset
-from flytekitplugins.spark import Spark
 
 try:
     from typing import Annotated
 except ImportError:
     from typing_extensions import Annotated
 
 # %% [markdown]
-# > We define two column types: `name: str` and `age: int`.
+# Create an `ImageSpec` to automate the retrieval of a prebuilt Spark image.
+# %%
+custom_image = ImageSpec(name="flyte-spark-plugin", registry="ghcr.io/flyteorg")
+
+# %% [markdown]
+# :::{note}
+# To upload the image to the local registry in the demo cluster, indicate the registry as `localhost:30000`.
+# :::
+#
+# The following imports are required to configure the Spark cluster in Flyte.
+# You can load them on demand.
+# %%
+if custom_image.is_container():
+    from flytekitplugins.spark import Spark
+
+# %% [markdown]
+# In this particular example,
+# we specify two column types: `name: str` and `age: int`
+# that we extract from the Spark DataFrame.
 # %%
 columns = kwtypes(name=str, age=int)
 
+
 # %% [markdown]
-# Next, we define a task that returns a Spark DataFrame.
+# To create a Spark task, add {py:class}`~flytekitplugins.spark.Spark` config to the Flyte task.
+#
+# The `spark_conf` parameter can encompass configuration choices commonly employed when setting up a Spark cluster.
+# Additionally, if necessary, you can provide `hadoop_conf` as an input.
+#
+# Create a task that yields a Spark DataFrame.
 # %%
 @task(
     task_config=Spark(
```

Comment:
I wonder how this works at registration time since the Spark plugin is only being imported `if custom_image.is_container()? Shouldn't the `spark_df` task be nested in that conditional?

Reply:
You're right. For now, I'm importing the library at the top-level.